### PR TITLE
support user profiles

### PR DESCRIPTION
**Pull request by Rizziepit**
- [x] add sessions
- [x] add authentication policy
- [x] use session as cache for profile data
- [x] do write-through on profile data update

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony ready for review. Depends on https://github.com/universalcore/unicore.hub.client/pull/5 though, because it needs a release

File: cms/views/cms_views.py [hunk 1]
```diff
@@ -318,3 +322,45 @@ def search(self):
...
+            try:
+                user = hubclient.get_user(
+                    ticket, self.request.route_url('redirect_to_login'))
+                self.request.session[USER_DATA_SESSION_KEY] = user.data
+                user_id = user.get('uuid')
+                headers = remember(self.request, user_id)
+                response.headerlist.extend(headers)
+
+            except HubClientException:
+                # TODO: what to do when ticket is invalid?
```
Comment (miltontony) [inline cms/views/cms_views.py]: probably show a retry link? With SSO I found that doing a logout then asking the user to login again as a good work around for invalid tickets.

Comment (Rizziepit) [inline cms/views/cms_views.py]: Is there a messaging module like Django's that can show a once-off error notification to the user?

Comment (miltontony) [inline cms/views/cms_views.py]: Hmm..don't know, haven't used one before

**Comment by miltontony** [comment]
This looks good. :+1: once travis passes

**Pull request merged**